### ISSUE TITLE
Expression diversity test case

### ISSUE THREAD

**Issue by carldlaird** [Pyomo/pyomo#259]
We should develop a Pyomo model that has a diverse set of expression object combinations so that it can be re-used in transformation tests, etc.

This model should contain (ideally) all the possible edge cases of expressions that we know about.

**Comment by blnicho** [comment]
We should add this before merging #272

**Comment by mrmundt** [comment]
@blnicho - #272 was merged. Is this still an issue?

**Comment by blnicho** [comment]
@mrmundt yes this is still an issue, it was not addressed in the linked PR
